### PR TITLE
[fix] EXPERIMENT: Switch docker base image to avoid mixing musl & glibc libraries at runtime

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

While testing Pulsar 4.0.0-preview.1, some compatibility issues with the Alpine base image have shown up.
It appears that the solution in the base image is something that will cause further compatibility issues. Mixing musl and glibc at runtime is known to cause compatibility and stability issues.

https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/24647#note_176723 states it clearly:
_"Combining glibc and musl runtimes is basically all but guaranteed to create an unstable environment, unless the system is appropriately configured (glibc side uses glibc binaries only, and vice versa)."_

https://github.com/sgerrand/alpine-pkg-glibc/issues/194#issuecomment-1510140972 mentions:
_"You really need everything pure musl, or everything musl (but works with gcompat or libc6-compat shim-like tools...) or everything glibc using this package)"_ 

In our case, we are mixing pure musl libraries and glibc compiled libraries with our Alpine base image configuration which should be avoided.

Another reason to switch to a base image that is optimized for running Java JVM applications. The current Alpine image crashes the JVM in unexpected ways. #23348/#23361 is an example of this.

```
java.lang.OutOfMemoryError: unable to create a native thread: possibly out of memory or process/resource limits reached.
```

There's also one more release blocker which could be due to the mixed musl & glibc environment:  #23306

### Why mixing musl & glibc at runtime should be avoided

* blog post by Ariadne Conill, [there is no such thing as a "glibc based alpine image"](https://ariadne.space/2021/08/26/there-is-no-such-thing-as-a-glibc-based-alpine-image/)
* comments explaining reasons:
  * https://github.com/docker-library/official-images/pull/10779#issuecomment-906095981
  * https://github.com/docker-library/official-images/pull/10779#issuecomment-906240756
  * https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/24647#note_176723

### Modifications

Use https://hub.docker.com/r/bellsoft/liberica-runtime-container as the base image. ([Apache NiFi uses a bellsoft/liberica-openjdk-debian](https://github.com/apache/nifi/blob/master/nifi-docker/dockerhub/Dockerfile#L20), so Liberica is already used in ASF.)
It's the Liberica OpenJDK's official base image provided by BellSoft. It uses Alpaquita Linux as the OS, which is similar to Alpine and uses `apk` as the package manager. Alpaquita Linux is optimized for running JVM applications.

The image size is very similar as before.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: docker/pulsar/Dockerfile [hunk 1]
```diff
@@ -116,31 +93,16 @@ prometheus_client\n\
...
-
-# Install GLibc compatibility library
-COPY --from=glibc /root/packages /root/packages
-RUN apk add --allow-untrusted --force-overwrite /root/packages/glibc-*.apk
-
-COPY --from=jvm /opt/jvm /opt/jvm
-ENV JAVA_HOME=/opt/jvm
-
-COPY --from=snappy-java /tmp/libsnappyjava.so /usr/lib/libsnappyjava.so
+" > /tmp/requirements.txt && RUN pip3 install --break-system-packages --no-cache-dir --only-binary grpcio -r /tmp/requirements.txt && rm /tmp/requirements.txt
```
Comment (nodece) [inline docker/pulsar/Dockerfile:96]:
```suggestion
" > /tmp/requirements.txt && pip3 install --break-system-packages --no-cache-dir --only-binary grpcio -r /tmp/requirements.txt && rm /tmp/requirements.txt
```

File: docker/pulsar/Dockerfile [hunk 2]
```diff
@@ -116,31 +93,16 @@ prometheus_client\n\
...
-RUN rm /requirements.txt
-
-# Install GLibc compatibility library
-COPY --from=glibc /root/packages /root/packages
-RUN apk add --allow-untrusted --force-overwrite /root/packages/glibc-*.apk
-
-COPY --from=jvm /opt/jvm /opt/jvm
-ENV JAVA_HOME=/opt/jvm
-
-COPY --from=snappy-java /tmp/libsnappyjava.so /usr/lib/libsnappyjava.so
```
Comment (nodece) [inline docker/pulsar/Dockerfile:131]: Why not remove the `libsnappyjava`? Is built-in snappy work fine?

Comment (lhotari) [reply docker/pulsar/Dockerfile:131]: Yes

Comment (lhotari) [reply docker/pulsar/Dockerfile:131]: The reason why snappy-java had to be built separately was due to the musl & glibc mixing issue.

**Comment by lhotari** [comment]
It looks like test image building fails due to missing supervisor package. It is available also with pip install: http://supervisord.org/installing.html

**Comment by lhotari** [comment]
> It looks like test image building fails due to missing supervisor package. It is available also with pip install: http://supervisord.org/installing.html

supervisor has been installed, but there is some remaining issue.

**Pull request closed without merging**